service/operator/service.go

```diff
@@ -123,6 +126,11 @@ func (s *Service) addFunc(obj interface{}) {
 		s.Logger.Log("error", "could not convert to azuretpr.CustomObject")
 	}
 
+	_, err := azure.NewClients(s.AzureConfig)
+	if err != nil {
+		s.Logger.Log("error", "could not create azure api clients '%#v'")
+	}
```

Comment:
IMO no. As mentioned above IMO the operator should use services to provide business logic. This business logic may use some client internally. Such clients should not be exposed outside the service though.